express/dataset.py

```diff
@@ -0,0 +1,209 @@
+"""This module defines the FondantDataset class, which is a wrapper around the manifest.
+
+It also defines the FondantComponent class, which uses the FondantDataset class to manipulate data.
+"""
+
+from abc import abstractmethod
+import argparse
+import json
+from pathlib import Path
+
+import dask.dataframe as dd
+
+from express.component_spec import ExpressComponent, kubeflow2python_type
+from express.manifest import Manifest, Subset, Index
+
+
+class FondantDataset:
+    """Wrapper around the manifest to download and upload data into a specific framework.
+
+    Uses Dask Dataframes for the moment.
+    """
+
+    def __init__(self, manifest):
+        self.manifest = manifest
+
+    @property
+    def project_name(self):
+        return self.manifest.project_name
+
+    def _load_subset(self, name: str) -> dd.DataFrame:
+        # get subset from the manifest
+        subset = self.manifest.get_subset(name)
+        # get its location and fields
+        # TODO remove gcp prefix
+        location = "gcs://" + subset.location + ".parquet"
+        fields = list(subset.fields.keys())
+
+        df = dd.read_parquet(
+            location, columns=fields, storage_options={"project": self.project_name}
+        )
+
+        return df
+
+    def load_data(self, component_spec: ExpressComponent) -> dd.DataFrame:
+        subsets = []
+        for name in component_spec.input_subsets.keys():
+            subset_data = self._load_subset(name)
+            subsets.append(subset_data)
+
+        # TODO this method should return a single dataframe with column_names called subset_field
+        # TODO add index
+        # dataset = concatenate_datasets(subsets)
+
+        # return dataset
+
+    def _upload_index(self, df) -> Index:
+        # get location
+        # TODO remove GCP prefix
+        remote_path = "gcs://" + self.manifest.index.location + ".parquet"
+        # upload to the cloud
+        dd.to_parquet(
+            df,
+            remote_path,
+            storage_options={"project": self.project_name},
+            overwrite=True,
+        )
+
+    def add_index(self, output_df):
+        index_columns = list(self.manifest.index.fields.keys())
+        # load index dataframe
+        index_df = output_df[index_columns]
+
+        self._upload_index(index_df)
+
+    def _upload_subset(self, name, fields, df) -> Subset:
+        # add subset to the manifest
+        fields = [(field.name, field.type) for field in fields.values()]
+        self.manifest.add_subset(name, fields=fields)
+        # upload to the cloud
+        # TODO remove prefix and suffix?
+        remote_path = "gcs://" + self.manifest.get_subset(name).location + ".parquet"
+        dd.to_parquet(
+            df,
```

Comment:
Shouldn't we remove the column name prefixes?

Reply:
You mean the subset name? That is already happening

Reply:
Can you point me to it?

Reply:
Sorry this was indeed not included yet, this is done now.